src/main/java/frc/robot/GenericPID/PathSegmentBase.java

```diff
@@ -0,0 +1,63 @@
+package frc.robot.GenericPID;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+
+public abstract class PathSegmentBase {
+    //IMPORTANT:
+    //assumes that the segment follows a function and can be chopped into smaller segments without superclass intervention
+    //therefore, structure your class so the members don't need to be changed based on the location of the segment.
+    public double x1;
+    public double x2;
+
+    public abstract double y(double x); //hehe2
+    
+    public PathSegmentBase chopEnd(double x) {
+        //          /
+        //        /
+        //      /
+        //    /
+        //  /
+        // |_ _ _ _|x
+        var s = duplicateWithSuperCopyConstructor();
+        s.x2 = x;
+        return s;
+    }
+    public PathSegmentBase chopBeginning(double x) {
+        //          /
+        //        /
+        //      /
+        //    /
+        //  /
+        //  x|_ _ _ _|
+        var s = duplicateWithSuperCopyConstructor();
+        s.x1 = x;
+        return s;
+    }
+    public ArrayList<PathSegmentBase> sliceChunk(double x1, double x2) {
+        //          /
+        //        /
+        //      /
+        //    /
+        //  /
+        // |_ _|x x|_|
+        ArrayList<PathSegmentBase> ret = new ArrayList<PathSegmentBase>();
+        ret.add(chopBeginning(x2));
+        ret.add(chopEnd(x1));
+        return ret;
+    }
+    
+    public abstract double derivative(double x);
+
+    private PathSegmentBase duplicateWithSuperCopyConstructor() {
+        try {
+            Class<? extends PathSegmentBase> sub = this.getClass(); 
```

Comment:
What are you trying to do here? Idk if the abstraction is necessary.

Reply:
whim bug fix, if you have something better you can use it, but what it does is copies the current object which is polymorphised into PathSegmentBase